packages/react-sdk/src/hooks/useStartConversation.ts

```diff
@@ -1,44 +1,40 @@
-import { useCallback, useContext } from "react";
+import { useCallback } from "react";
 import type { SendOptions, InvitationContext } from "@xmtp/xmtp-js";
-import { XMTPContext } from "../contexts/XMTPContext";
+import { useClient } from "./useClient";
 
 /**
  * This hook starts a new conversation and sends an initial message to it.
  */
 export const useStartConversation = <T = string>(
   options?: InvitationContext,
 ) => {
-  const xmtpContext = useContext(XMTPContext);
-  if (xmtpContext === undefined) {
-    console.error("useStartConversation must be used within a XMTPProvider");
-  }
+  const { client } = useClient();
 
   // destructure options for more granular dependency arrays
   const { conversationId, metadata } = options ?? {};
 
   return useCallback(
     async (peerAddress: string, message: T, sendOptions?: SendOptions) => {
       // we can't do anything without a client
-      if (xmtpContext?.client === undefined) {
+      if (client === undefined) {
         console.error("XMTP client is not available");
         return undefined;
       }
 
-      const conversation =
-        await xmtpContext?.client?.conversations.newConversation(
-          peerAddress,
-          conversationId && metadata
-            ? {
-                conversationId,
-                metadata,
-              }
-            : undefined,
-        );
+      const conversation = await client?.conversations.newConversation(
+        peerAddress,
+        conversationId && metadata
+          ? {
+              conversationId,
+              metadata,
```

Comment:
I know the SDK requires you to have metadata as an empty object when you have none. We could totally pave over that here and just set to `{}` if a conversation ID is specified but metadata is not.